src/request.ts

```diff
@@ -1111,12 +1128,12 @@ class DatastoreRequest {
    *
    * @private
    */
-  request_(config, callback) {
+  request_(config: RequestConfig): Promise<RequestResponse>;
+  request_(config: RequestConfig, callback: RequestCallback): void;
+  request_(config: RequestConfig, callback: RequestCallback = (() => {})): void|Promise<RequestResponse> {
```

Comment:
I don't think you want to default the callback to `(() => {}))`, and it should be marked as optional here.

Reply:
Oh wait!  This is the `request_` method.  I don't think we want the promise overloads here at all.  I *think* the promisification of this gets filtered out.

Reply:
Beautiful, I deleted the default callback and the promise overloads